resolver/bootstrap.go

```diff
@@ -134,7 +142,13 @@ func (b *Bootstrap) resolveUpstream(r Resolver, host string) ([]net.IP, error) {
 // NewHTTPTransport returns a new http.Transport that uses b to resolve hostnames
 func (b *Bootstrap) NewHTTPTransport() *http.Transport {
 	if b.resolver == nil {
-		return &http.Transport{}
+		dialer := &net.Dialer{
+			Timeout: defaultTimeout,
+		}
+
+		return &http.Transport{
+			DialContext: dialer.DialContext,
+		}
```

Comment:
Let's re-use `b.dialer` so we also use `cfg.Upstreams.Timeout` if it is set:
```suggestion
		return &http.Transport{
			DialContext: b.dialer.DialContext,
		}
```

Reply:
:+1: